From 7a654e3796602a21c0e8cd93f12e16446ea26678 Mon Sep 17 00:00:00 2001
From: Fabian Maurer <dark.shadow4@web.de>
Date: Sat, 28 Jul 2018 16:31:46 +0200
Subject: [PATCH 12/24] ntoskrnl: Emulate sti/cli instructions on x86_64

Fixes bug 45521.
Thanks to Anastasius Focht for the clear bug report.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=45521
Signed-off-by: Fabian Maurer <dark.shadow4@web.de>
---
 dlls/ntoskrnl.exe/instr.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/ntoskrnl.exe/instr.c b/dlls/ntoskrnl.exe/instr.c
index b2dac58..67cdd32 100644
--- a/dlls/ntoskrnl.exe/instr.c
+++ b/dlls/ntoskrnl.exe/instr.c
@@ -815,6 +815,11 @@ static DWORD emulate_instruction( EXCEPTION_RECORD *rec, CONTEXT *context )
         }
         break;  /* Unable to emulate it */
     }
+
+    case 0xfa: /* cli */
+    case 0xfb: /* sti */
+        context->Rip += prefixlen + 1;
+        return ExceptionContinueExecution;
     }
     return ExceptionContinueSearch;  /* Unable to emulate it */
 }
-- 
1.9.1

